core/src/avm2/globals/flash/net/sharedobject.rs

```diff
@@ -39,28 +47,444 @@ fn class_init<'gc>(
     Ok(Value::Undefined)
 }
 
-fn get_local<'gc>(
+/// Serialize a Value to an AmfValue
+fn serialize_value<'gc>(
     activation: &mut Activation<'_, 'gc, '_>,
-    _this: Option<Object<'gc>>,
-    _args: &[Value<'gc>],
+    elem: Value<'gc>,
+) -> Option<AmfValue> {
+    match elem {
+        Value::Undefined => Some(AmfValue::Undefined),
+        Value::Null => Some(AmfValue::Null),
+        Value::Bool(b) => Some(AmfValue::Bool(b)),
+        Value::Number(f) => Some(AmfValue::Number(f)),
+        Value::Integer(num) => {
+            if num >= (1 << 28) || num <= -(1 << 28) {
```

Comment:
Note: AFAIK the "real" logic here should be that ints above/below 1<<28 are immediately,  implicitly converted to float (Number), because avmplus's int atom representation just can't represent them; as in, such a Value::Integer should never be encountered.
I'm not saying to do anything about it in this PR, but maybe let's add a comment around logic like this, saying that long-term it shouldn't be needed.

Reply:
I've added a comment stating that the `Value::Unsigned/Value::Integer` to `Value::Number` conversion should be done separately from serialization.